File: src/public/cognitoAdmin/deleteUser.js
"use strict";

const { requireSuperAdmin } = require("../../common/auth");

async function handleCognitoDeleteUser({ res, auth, params }) {
  const okRes = requireSuperAdmin({ auth, res });
  if (okRes !== true) return okRes;

  return res.notImplemented(`TODO: delete Cognito user ${params.username}`);
}

module.exports = { handleCognitoDeleteUser };

File: src_C/Public/CognitoAdmin/DeleteUser.cs
using Amazon.Lambda.APIGatewayEvents;
using RecallSmith.Lambda.Common;

namespace RecallSmith.Lambda.Public.CognitoAdmin;

public static class DeleteUser
{
  public static Task<APIGatewayProxyResponse> HandleCognitoDeleteUser(
    LambdaRequest req,
    Res res,
    AuthContext auth,
    IReadOnlyDictionary<string, string> parameters)
  {
    var deny = Auth.RequireSuperAdmin(auth, res);
    if (deny is not null) return Task.FromResult(deny);

    var username = parameters.TryGetValue("username", out var u) ? u : string.Empty;
    return Task.FromResult(res.NotImplemented($"TODO: delete Cognito user {username}"));
  }
}

